src/Core/Services/Implementations/UserService.cs

```diff
@@ -223,59 +223,39 @@ public async Task SaveUserAsync(User user, bool push = false)
 
     public override async Task<IdentityResult> DeleteAsync(User user)
     {
-        // Check if user is the only owner of any organizations.
-        var onlyOwnerCount = await _organizationUserRepository.GetCountByOnlyOwnerAsync(user.Id);
-        if (onlyOwnerCount > 0)
-        {
-            var deletedOrg = false;
-            var orgs = await _organizationUserRepository.GetManyDetailsByUserAsync(user.Id,
-                OrganizationUserStatusType.Confirmed);
-            if (orgs.Count == 1)
-            {
-                var org = await _organizationRepository.GetByIdAsync(orgs.First().OrganizationId);
-                if (org != null && (!org.Enabled || string.IsNullOrWhiteSpace(org.GatewaySubscriptionId)))
-                {
-                    var orgCount = await _organizationUserRepository.GetCountByOrganizationIdAsync(org.Id);
-                    if (orgCount <= 1)
-                    {
-                        await _organizationRepository.DeleteAsync(org);
-                        deletedOrg = true;
-                    }
-                }
-            }
+        var userList = await ValidateDeleteUsersAsync(new List<User> { user });
+        var results = userList.ToList()[0].ErrorMessage;
 
-            if (!deletedOrg)
-            {
-                return IdentityResult.Failed(new IdentityError
-                {
-                    Description = "Cannot delete this user because it is the sole owner of at least one organization. Please delete these organizations or upgrade another user.",
-                });
-            }
+        if (results == string.Empty || results == null)
+        {
+            await _userRepository.DeleteAsync(user);
+            await _referenceEventService.RaiseEventAsync(
+                new ReferenceEvent(ReferenceEventType.DeleteAccount, user, _currentContext));
+            await _pushService.PushLogOutAsync(user.Id);
+            return IdentityResult.Success;
         }
-
-        var onlyOwnerProviderCount = await _providerUserRepository.GetCountByOnlyOwnerAsync(user.Id);
-        if (onlyOwnerProviderCount > 0)
+        else
         {
             return IdentityResult.Failed(new IdentityError
             {
-                Description = "Cannot delete this user because it is the sole owner of at least one provider. Please delete these providers or upgrade another user.",
+                Description = results
             });
         }
+    }
+
+    public async Task<IEnumerable<(Guid UserId, string? ErrorMessage)>> DeleteManyAsync(IEnumerable<User> users)
+    {
+        var results = await ValidateDeleteUsersAsync(users.ToList());
 
-        if (!string.IsNullOrWhiteSpace(user.GatewaySubscriptionId))
+        await _userRepository.DeleteManyAsync(users);
```

Comment:
You should only delete users that have passed validation. Something like
```suggestion
        var usersToDelete = users.Where(u => string.IsNullOrEmpty(u.ErrorMessage);
        await _userRepository.DeleteManyAsync(usersToDelete);
        foreach (var user in usersToDelete)
```